modules/e2ee/Context.js

```diff
@@ -282,9 +329,12 @@ export class Context {
             // Verify the long-term signature of the authentication tag.
             if (signatureLength) {
```

Comment:
if we don't get a signature we need to check whether we should have received one or set the status to "not trustworthy"

Reply:
also if we get a valid signature we can now determine a ratio of valid past frames. If that is too low...